solr/solrj/src/java/org/apache/solr/common/util/Utils.java

```diff
@@ -302,10 +302,16 @@ public static byte[] toUTF8(CharArr out) {
   }
 
   public static Object fromJSON(byte[] utf8) {
+    if (utf8 == null || utf8.length == 0) {
+      return Collections.emptyMap();
+    }
```

Comment:
Need in both fromJSON methods since utf8.length returned NPE without this.

Reply:
Add this comment to the code, somebody will be looking at this in five years and be confused

Reply:
Fixed in 2011fb0c51476478322e41a4f2b5259f44ea6662